From 624d523361dcf8c10742cbcb1a017e21675002ff Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Sat, 22 Sep 2012 02:17:08 -0400
Subject: [PATCH] Replace Fedora defaults with Gentoo ones, and disable code
 not relevant for Gentoo

Disable bugzilla and ureport plugins (they require Gentoo infrastructure
changes). Fix libreport-web.pc to not rely on xmlrpc.pc, which Gentoo
does not have.

Disble mailx post-create by default to avoid "abrtd: 'post-create' ...
exited with 141" errors.
---
 doc/Makefile.am                    |  2 -
 doc/reporter-bugzilla.txt          |  2 +-
 libreport-web.pc.in                |  6 +--
 src/lib/dump_dir.c                 |  2 +-
 src/plugins/Makefile.am            | 83 ++------------------------------------
 src/plugins/bugzilla.conf          |  2 +-
 src/plugins/mailx_event.conf       |  2 +-
 src/plugins/report_Bugzilla.conf   |  2 +-
 src/plugins/report_Bugzilla.xml.in |  4 +-
 src/plugins/reporter-bugzilla.c    |  2 +-
 src/report-python/__init__.py      |  4 +-
 11 files changed, 16 insertions(+), 95 deletions(-)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index 542fe4c..863b6d1 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -9,11 +9,9 @@ XMLTO_SILENT_0 = @echo "  XMLTO " $@;
 
 MAN1_TXT =
 MAN1_TXT += report-cli.txt
-MAN1_TXT += reporter-bugzilla.txt
 MAN1_TXT += reporter-kerneloops.txt
 MAN1_TXT += reporter-mailx.txt
 MAN1_TXT += reporter-print.txt
-MAN1_TXT += reporter-rhtsupport.txt
 MAN1_TXT += reporter-upload.txt
 MAN1_TXT += report.txt
 
diff --git a/doc/reporter-bugzilla.txt b/doc/reporter-bugzilla.txt
index 3cedf73..734d598 100644
--- a/doc/reporter-bugzilla.txt
+++ b/doc/reporter-bugzilla.txt
@@ -44,7 +44,7 @@ Configuration file lines should have 'PARAM = VALUE' format. The parameters are:
 	Password to Bugzilla account.
 
 'BugzillaURL'::
-	Bugzilla HTTP(S) address. (default: https://bugzilla.redhat.com)
+	Bugzilla HTTP(S) address. (default: https://bugs.gentoo.org/)
 
 'SSLVerify'::
 	Use yes/true/on/1 to verify server's SSL certificate. (default: yes)
diff --git a/libreport-web.pc.in b/libreport-web.pc.in
index c9af1d2..14505f8 100644
--- a/libreport-web.pc.in
+++ b/libreport-web.pc.in
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: libreport
 Description: Library providing network API for libreport
 Version: @VERSION@
-Requires: glib-2.0 libcurl libproxy-1.0 libxml-2.0 xmlrpc xmlrpc_client json btparser libreport
-Libs: -L${libdir} -lreport-web
-Cflags:
+Requires: glib-2.0 libcurl libproxy-1.0 libxml-2.0 json btparser libreport
+Libs: -L${libdir} @XMLRPC_LIBS@ @XMLRPC_CLIENT_LIBS@ -lreport-web
+Cflags: @XMLRPC_CFLAGS@ @XMLRPC_CLIENT_CFLAGS@
 
diff --git a/src/lib/dump_dir.c b/src/lib/dump_dir.c
index 0fc7da7..41e4f55 100644
--- a/src/lib/dump_dir.c
+++ b/src/lib/dump_dir.c
@@ -556,7 +556,7 @@ void dd_create_basic_files(struct dump_dir *dd, uid_t uid, const char *chroot_di
         release = load_text_file("/etc/system-release",
                 DD_LOAD_TEXT_RETURN_NULL_ON_FAILURE);
         if (!release)
-            release = load_text_file("/etc/redhat-release", /*flags:*/ 0);
+            release = load_text_file("/etc/gentoo-release", /*flags:*/ 0);
         dd_save_text(dd, FILENAME_OS_RELEASE, release);
         if (chroot_dir)
         {
diff --git a/src/plugins/Makefile.am b/src/plugins/Makefile.am
index 967596f..3f9dfd7 100644
--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -1,36 +1,27 @@
 pluginslibdir = $(PLUGINS_LIB_DIR)
 
 bin_PROGRAMS = \
-    reporter-bugzilla \
-    reporter-rhtsupport \
     reporter-kerneloops \
     reporter-upload \
     reporter-mailx \
     reporter-print \
-    report \
-    reporter-ureport
+    report
 
 pluginsconfdir = $(PLUGINS_CONF_DIR)
 
 reportpluginsconfdir = $(REPORT_PLUGINS_CONF_DIR)
 
 dist_reportpluginsconf_DATA = \
-    bugzilla.conf \
-    rhtsupport.conf \
     mailx.conf
 
 eventsdir = $(EVENTS_DIR)
 
 dist_events_DATA = \
-    report_Bugzilla.xml \
-    report_Bugzilla.conf \
     report_Logger.conf \
     report_Logger.xml \
     report_Mailx.xml \
-    report_RHTSupport.xml \
     report_Kerneloops.xml \
-    report_Uploader.xml \
-    report_uReport.xml
+    report_Uploader.xml
 
 @INTLTOOL_XML_RULE@
 
@@ -39,74 +30,18 @@ eventsconfdir = $(EVENTS_CONF_DIR)
 dist_eventsconf_DATA = \
     mailx_event.conf \
     print_event.conf \
-    bugzilla_event.conf \
-    rhtsupport_event.conf \
     uploader_event.conf
 
 EXTRA_DIST = \
-    report_Bugzilla.xml.in \
-    report_Bugzilla.conf \
     report_Logger.conf \
     report_Logger.xml.in \
     report_Mailx.xml.in \
-    report_RHTSupport.xml.in \
     report_Kerneloops.xml.in \
-    report_Uploader.xml.in \
-    report_uReport.xml.in
+    report_Uploader.xml.in
 
 $(DESTDIR)/$(DEBUG_INFO_DIR):
 	$(mkdir_p) '$@'
 
-reporter_bugzilla_SOURCES = \
-    reporter-bugzilla.c rhbz.c rhbz.h
-reporter_bugzilla_CPPFLAGS = \
-    -I$(srcdir)/../include \
-    -I$(srcdir)/../lib \
-    -DBIN_DIR=\"$(bindir)\" \
-    -DCONF_DIR=\"$(CONF_DIR)\" \
-    -DLOCALSTATEDIR='"$(localstatedir)"' \
-    -DDEBUG_DUMPS_DIR=\"$(DEBUG_DUMPS_DIR)\" \
-    -DDEBUG_INFO_DIR=\"$(DEBUG_INFO_DIR)\" \
-    -DPLUGINS_LIB_DIR=\"$(PLUGINS_LIB_DIR)\" \
-    -DPLUGINS_CONF_DIR=\"$(REPORT_PLUGINS_CONF_DIR)\" \
-    $(GLIB_CFLAGS) \
-    $(LIBREPORT_CFLAGS) \
-    $(XMLRPC_CFLAGS) $(XMLRPC_CLIENT_CFLAGS) \
-    -D_GNU_SOURCE
-reporter_bugzilla_LDADD = \
-    $(GLIB_LIBS) \
-    $(XMLRPC_LIBS) $(XMLRPC_CLIENT_LIBS) \
-    ../lib/libreport-web.la \
-    ../lib/libreport.la
-
-reporter_rhtsupport_SOURCES = \
-    abrt_rh_support.h abrt_rh_support.c \
-    reporter-rhtsupport.h \
-    reporter-rhtsupport-parse.c \
-    reporter-rhtsupport.c
-reporter_rhtsupport_CPPFLAGS = \
-    -I$(srcdir)/../include \
-    -I$(srcdir)/../lib \
-    -DBIN_DIR=\"$(bindir)\" \
-    -DCONF_DIR=\"$(CONF_DIR)\" \
-    -DLOCALSTATEDIR='"$(localstatedir)"' \
-    -DDEBUG_DUMPS_DIR=\"$(DEBUG_DUMPS_DIR)\" \
-    -DDEBUG_INFO_DIR=\"$(DEBUG_INFO_DIR)\" \
-    -DPLUGINS_LIB_DIR=\"$(PLUGINS_LIB_DIR)\" \
-    -DPLUGINS_CONF_DIR=\"$(REPORT_PLUGINS_CONF_DIR)\" \
-    $(GLIB_CFLAGS) \
-    $(LIBREPORT_CFLAGS) \
-    $(XMLRPC_CFLAGS) $(XMLRPC_CLIENT_CFLAGS) \
-    $(LIBXML_CFLAGS) \
-    -D_GNU_SOURCE
-reporter_rhtsupport_LDFLAGS = -ltar
-reporter_rhtsupport_LDADD = \
-    $(GLIB_LIBS) \
-    $(LIBXML_LIBS) \
-    $(XMLRPC_LIBS) $(XMLRPC_CLIENT_LIBS) \
-    ../lib/libreport-web.la \
-    ../lib/libreport.la
-
 reporter_upload_SOURCES = \
     reporter-upload.c
 reporter_upload_CPPFLAGS = \
@@ -200,16 +135,4 @@ report_CPPFLAGS = \
 report_LDADD = \
     ../lib/libreport.la
 
-reporter_ureport_SOURCES = \
-    ureport.c
-reporter_ureport_CPPFLAGS = \
-    -I$(srcdir)/../include \
-    -I$(srcdir)/../lib \
-    $(GLIB_CFLAGS) \
-    $(LIBREPORT_CFLAGS) \
-    -D_GNU_SOURCE
-reporter_ureport_LDADD = \
-    ../lib/libreport.la \
-    ../lib/libreport-web.la
-
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
diff --git a/src/plugins/bugzilla.conf b/src/plugins/bugzilla.conf
index 18eba5a..f5afe6f 100644
--- a/src/plugins/bugzilla.conf
+++ b/src/plugins/bugzilla.conf
@@ -1,5 +1,5 @@
 # Bugzilla URL
-BugzillaURL = https://bugzilla.redhat.com/
+BugzillaURL = https://bugs.gentoo.org/
 # yes means that ssl certificates will be checked
 SSLVerify = yes
 # your login has to exist, if you don have any, please create one
diff --git a/src/plugins/mailx_event.conf b/src/plugins/mailx_event.conf
index a8c5d11..1c07a73 100644
--- a/src/plugins/mailx_event.conf
+++ b/src/plugins/mailx_event.conf
@@ -1,3 +1,3 @@
-EVENT=post-create reporter-mailx --notify-only -c /etc/libreport/plugins/mailx.conf
+# EVENT=post-create reporter-mailx --notify-only -c /etc/libreport/plugins/mailx.conf
 
 EVENT=report_Mailx      reporter-mailx
diff --git a/src/plugins/report_Bugzilla.conf b/src/plugins/report_Bugzilla.conf
index f1a77f5..bb2e651 100644
--- a/src/plugins/report_Bugzilla.conf
+++ b/src/plugins/report_Bugzilla.conf
@@ -1,4 +1,4 @@
-Bugzilla_BugzillaURL = https://bugzilla.redhat.com
+Bugzilla_BugzillaURL = https://bugs.gentoo.org/
 Bugzilla_Login =
 Bugzilla_Password =
 Bugzilla_SSLVerify = yes
diff --git a/src/plugins/report_Bugzilla.xml.in b/src/plugins/report_Bugzilla.xml.in
index 8151d32..acaf7fb 100644
--- a/src/plugins/report_Bugzilla.xml.in
+++ b/src/plugins/report_Bugzilla.xml.in
@@ -16,8 +16,8 @@
             <_label>Bugzilla URL</_label>
             <allow-empty>no</allow-empty>
             <_description>Address of Bugzilla server</_description>
-            <default-value>https://bugzilla.redhat.com</default-value>
-            <_note-html>You can create bugzilla.redhat.com account &lt;a href="https://bugzilla.redhat.com/createaccount.cgi"&gt;here&lt;/a&gt;</_note-html>
+            <default-value>https://bugs.gentoo.org/</default-value>
+            <_note-html>You can create bugs.gentoo.org account &lt;a href="https://bugs.gentoo.org/createaccount.cgi"&gt;here&lt;/a&gt;</_note-html>
         </option>
         <option type="text" name="Bugzilla_Login">
             <_label>User name</_label>
diff --git a/src/plugins/reporter-bugzilla.c b/src/plugins/reporter-bugzilla.c
index a3e8855..17a686d 100644
--- a/src/plugins/reporter-bugzilla.c
+++ b/src/plugins/reporter-bugzilla.c
@@ -34,7 +34,7 @@ static void set_settings(struct bugzilla_struct *b, map_string_h *settings)
     environ = getenv("Bugzilla_BugzillaURL");
     b->b_bugzilla_url = environ ? environ : get_map_string_item_or_empty(settings, "BugzillaURL");
     if (!b->b_bugzilla_url[0])
-        b->b_bugzilla_url = "https://bugzilla.redhat.com";
+        b->b_bugzilla_url = "https://bugs.gentoo.org/";
     b->b_bugzilla_xmlrpc = xasprintf("%s"XML_RPC_SUFFIX, b->b_bugzilla_url);
 
     environ = getenv("Bugzilla_SSLVerify");
diff --git a/src/report-python/__init__.py b/src/report-python/__init__.py
index 9c5e6b3..7188392 100644
--- a/src/report-python/__init__.py
+++ b/src/report-python/__init__.py
@@ -21,8 +21,8 @@ from report.io import TextIO, GTKIO, NewtIO
 
 import os
 
-SYSTEM_RELEASE_PATHS = ["/etc/system-release","/etc/redhat-release"]
-SYSTEM_RELEASE_DEPS = ["system-release", "redhat-release"]
+SYSTEM_RELEASE_PATHS = ["/etc/system-release","/etc/gentoo-release"]
+SYSTEM_RELEASE_DEPS = ["system-release", "gentoo-release"]
 
 _hardcoded_default_product = ""
 _hardcoded_default_version = ""
-- 
1.7.12

